### PR TITLE
Validation for Asset precision 

### LINKED ISSUE #531: Validation for register trading pair
Write `ensure!` to make sure the correct precision is being configured for a given trading pair

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR verifies if `scale` of `price_tick_size`  is less than or equal to `quote_asset_precision`

**DO NOT MERGE** 
There is a `polkadex-primitives` PR that needs to be merged before we merge this PR.

### CODE REVIEW

File: pallets/ocex/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ frame-support = { default-features = false, branch = "polkadot-v0.9.28", git = '
 frame-system = { default-features = false, branch = "polkadot-v0.9.28", git = 'https://github.com/paritytech/substrate.git' }
 sp-runtime = { default-features = false, branch = "polkadot-v0.9.28", git = 'https://github.com/paritytech/substrate.git' }
 sp-std = { default-features = false, branch = "polkadot-v0.9.28", git = 'https://github.com/paritytech/substrate.git' }
-polkadex-primitives = { default-features = false, branch = "polkadot-v0.9.28", git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git" }
+polkadex-primitives = { default-features = false, branch = "fx-precisions-serde-fix", git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git" }
```
Comment (serhii-temchenko) [inline pallets/ocex/Cargo.toml]: Please revert

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -361,6 +363,8 @@ pub mod pallet {
 			max_order_qty: BalanceOf<T>,
 			price_tick_size: BalanceOf<T>,
 			qty_step_size: BalanceOf<T>,
+			base_asset_precision: u8,
```
Comment (Gauthamastro) [inline pallets/ocex/src/lib.rs]: We don't need this input from the user, we can calculate it from scale()

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/ocex/Cargo.toml (hunk 1 above)
Comment (felixfaisal) [reply pallets/ocex/Cargo.toml]: Updated ✅

File: pallets/ocex/src/lib.rs (hunk 2 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Updated ✅

**Comment by serhii-temchenko** [comment]
CI is failing because of outdated reference in the lock file, please update it.

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**